db/event_manager_test.go

```diff
@@ -275,7 +277,52 @@ func TestUnhandledEvent(t *testing.T) {
 
 }
 
-func InitWebhookTest() (*int, *float64, *[][]byte) {
+func GetRouterWithHandler(counter *int, sum *float64, payloads *[][]byte) http.Handler {
+	r := http.NewServeMux()
+	r.HandleFunc("/slow", func(w http.ResponseWriter, r *http.Request) {
+		time.Sleep(1000 * time.Millisecond)
+		*counter++
+		fmt.Fprintf(w, "OK")
+	})
+	r.HandleFunc("/slow_2s", func(w http.ResponseWriter, r *http.Request) {
+		time.Sleep(2 * time.Second)
+		*counter++
+		fmt.Fprintf(w, "OK")
+	})
+	r.HandleFunc("/slow_5s", func(w http.ResponseWriter, r *http.Request) {
+		time.Sleep(5 * time.Second)
+		*counter++
+		fmt.Fprintf(w, "OK")
+	})
+	r.HandleFunc("/", func(w http.ResponseWriter, r *http.Request) {
+		r.ParseForm()
+		body, err := ioutil.ReadAll(r.Body)
+		if err != nil {
+			log.Printf("Error trying to read body: %s", err)
+		}
+		if len(body) > 0 {
+			*payloads = append(*payloads, body)
```

Comment:
One of the unit tests is failing with a data race here.  It feels like we need to move counter/sum/payloads to a struct with a mutex to manage concurrent interaction.  A single mutex to manage access to all three is probably sufficient - I don't think we need property level granularity. 

That would also let us get rid of some of the slice reference ugliness going on here.